src/borda_clj/client.clj

```diff
@@ -0,0 +1,91 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(def submit-key {:op "_submit"})
+
+(defn submit-failed [measurements values]
+  (update measurements submit-key (partial merge-with +) {:error_count (get values :_submits)}))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and
+   caps the size of the hashmap at the given max-buffer-size."
+  [measurements max-buffer-size dimensions values]
+  (let [key  (into (sorted-map) dimensions)
+        vals (update values :_submits (fnil max 0 0) 1)] ; use existing _points if available, else 1
```

Comment:
Since you're providing the second argument to `max` as a constant 1, you don't need the second 0 in the `fnil` call.

Reply:
A more succint way to say this, though, would be `(let [... ... vals (merge {:_submits 1} values)] ...)`.